contracts/interchain-token/HyperliquidInterchainTokenService.sol

```diff
@@ -49,15 +64,25 @@ contract HyperLiquidInterchainTokenService is InterchainTokenService {
     /**
      * @notice Updates the deployer for a specific Hyperliquid token
      * @dev Only callable by the operator or owner for security. This keeps permissioning logic in the service layer
-     * @param token The Hyperliquid token contract
+     * @param tokenId The tokenId of the Hyperliquid token
      * @param newDeployer The new deployer address
      */
-    function updateTokenDeployer(IHyperliquidDeployer token, address newDeployer) external onlyOperatorOrOwner {
-        // Additional validation: ensure the token is a valid Hyperliquid token
-        // This could be enhanced with a registry check if needed
-        if (address(token) == address(0)) revert InvalidTokenAddress();
+    function updateTokenDeployer(bytes32 tokenId, address newDeployer) external onlyOperatorOrOwner {
+        address tokenManagerAddr = tokenManagerAddress(tokenId);
+        if (tokenManagerAddr == address(0)) revert TokenManagerDoesNotExist(tokenId);
+
+        address tokenAddress = registeredTokenAddress(tokenId);
+        if (tokenAddress == address(0)) revert ZeroAddress();
+
+        if (!_supportsHyperliquidInterface(tokenAddress)) {
+            revert TokenDoesNotSupportHyperliquidInterface(tokenAddress);
+        }
+
```

Comment:
```suggestion
```
not needed (also if you wanted to check this, you'd just cast it and call deployer directly IHyperliquidDeployer(tokenAddress).deployer()` instead of a low level static call)

Calling `updateDeployer` already checks that it supports the interface

Reply:
this was not addressed, not sure why it was marked as resolved